### ISSUE TITLE
Type inference appears to be broken - unsure of expected behavior

### ISSUE THREAD

**Issue by arcticmatt** [graphql-nexus/schema#507]
I'm adding this code to https://github.com/graphql-nexus/schema/blob/18a306ee2903d98fc110c1067e1903ea0ee5f256/examples/star-wars/src/graphql/query.ts after cloning the repo and running `yarn`. 

```
export const Foo = objectType({
  name: "Query",
  definition(t) {
    t.field("string", {
      type: "Int",
      // Why is this NOT a type error?
      resolve: () => "string not int",
    });
    t.field("foo", {
      type: "Droid",
      args: characterArgs,
      // Why is this NOT a type error?
      resolve: () => 5,
    });
    t.field("droid", {
      type: "Droid",
      args: characterArgs,
      // Why is this a type error?
      resolve: () => 5,
    });
    t.int("bar", {
      // Why is this NOT a type error?
      resolve: () => "string not int",
    });
  },
});
```

In general, I don't really get how the type checking works. Why are those three issues not caught? 
1. I'm returning a string from `resolve` even though type is `"Int"`.
2. This is exactly the same as the third field, except the name of the field is different... but it passes the type checker.
3. For some reason this throws a type error, I don't get why.
4. Same as above but using `int` instead of `field`, I'm returning a string from `resolve` even though type is `"Int"`.

Is there any documentation I can refer to about how this works (from looking through the docs, doesn't look like it)? The documentation makes things sound fairly simple, but after spending a decent amount of time playing around with the library, I am very confused about how things are type checked.

*Note*: I can work around this by explicitly typing everything, i.e. doing something like
```
t.field("string", {
  type: "Int",
  resolve: (): number => "string not int",
});
t.field("foo", {
  type: "Droid",
  args: characterArgs,
  resolve: (): NexusGenRootTypes["Droid"] => 5,
});
```

However, I'm not sure if this is the recommended way to do things. From browsing around, it seems like Nexus *should* be able to do some automatic type inference.

cc @tgriesser @jasonkuhrt

**Comment by Weakky** [comment]
Hey @arcticmatt,

I believe you forgot to run the server. Nexus schema generates typegen at runtime. If you do not run it, there's not much we can infer. Here's your snippet after running the server:

![image](https://user-images.githubusercontent.com/25233379/95754306-20893e80-0ca3-11eb-8b4b-c007563fc352.png)

Closing now, let me know if you still encounter issues.

**Comment by Sytten** [comment]
Related to #418
